dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -814,7 +814,7 @@ HRESULT CDefaultContextMenu::DoRename(LPCMINVOKECOMMANDINFOEX lpcmi)
     CComPtr<IShellBrowser> psb;
     HRESULT hr;
 
-    if (!m_site || !m_cidl)
+    if (!m_site || m_cidl == 0)
```

Comment:
UINT m_cidl;
is the type of m_cidl.
I would be very surprised if your change really did change the behavior in any way. To me it looks as if that new code did the same as before your change.

Reply:
@JoachimHenze  I was actually surprised when I found that the "!m_cidl" test returned FALSE (its m_cidl value being 1). This is why DoRename returned E_FAIL.
After the change presented, DoRename returns S_OK;

Reply:
Please compile and run mi changes to comfirm please.